pkg/datapath/linux/node.go

```diff
@@ -509,6 +509,14 @@ func upsertIPsecLog(err error, spec string, loc, rem *net.IPNet, spi uint8) {
 	})
 	if err != nil {
 		scopedLog.WithError(err).Error("IPsec enable failed")
+
+		for _, acc := range accs {
+			*acc = errors.Join(*acc, fmt.Errorf("create endpoint %s (local_ip=%s, rem=%s, spi=%d): %w",
```

Comment:
Good point, the goal here was to consume as part of module health status, but in the meantime I have added extra logging and the node manager level so this could produce some ugly logs (I believe the newlines would be escaped by logrus though?). 

Wrapping doesn't really make sense in this context since the %w directive is supposed to add more context on top of an existing error, whereas the goal here is to provide a list of errors to indicate where reconciliation procedure calls have failed.

Ultimately, we do want to avoid this accumulation logic and just have well defined procedures that return when they encounter an error, thus reducing the potential number of states massively.  But I want to do more discussion/proposal work for that first.

Reply:
> (I believe the newlines would be escaped by logrus though?).

TBH, I'm not sure.

> Wrapping doesn't really make sense in this context since the %w directive is supposed to add more context on top of an existing error, whereas the goal here is to provide a list of errors to indicate where reconciliation procedure calls have failed.

Actually, since Go 1.20 you can use `fmt.Errorf` with multiple `%w` directives, just to wrap multiple errors and customize the wrapped error string representation, like this:

`fmt.Errorf("%w-%w", err1, err2)`

That's basically equivalent of just wrapping `err1` and `err2` with `errors.Join`, but with the additional possibility of getting rid of the newline and specify a custom separator.

From the Go 1.20 release notes:

> The [fmt.Errorf](https://tip.golang.org/pkg/fmt/#Errorf) function now supports multiple occurrences of the %w format verb, which will cause it to return an error that wraps all of those error operands.
>
> The new function [errors.Join](https://tip.golang.org/pkg/errors/#Join) returns an error wrapping a list of errors.

Relevant code for [fmt.Errorf](https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/fmt/errors.go) and for [errors.Join](https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/fmt/errors.go).

> But I want to do more discussion/proposal work for that first.

Sounds good, then, let's go with this logic for now :+1: